hotxlfp/grammarparser/parser.py

```diff
@@ -76,6 +76,9 @@ def p_expression_arithmetic_operator(self, p):
         """
         expression : expression PLUS expression
                   | expression MINUS expression
+                  | expression_paren MINUS expression_paren
+                  | expression MINUS expression_paren
+                  | expression_paren MINUS expression
```

Comment:
This gives the precedence over reducing to the `implicit_multiplication` handler.

With this change:
```
State  : 4
Stack  : expression_paren . LexToken(MINUS,'-',1,13)
Action : Shift and goto state 33

State  : 33
Stack  : expression_paren MINUS . LexToken(NUMBER,'1',1,15)
Action : Shift and goto state 5

State  : 5
Stack  : expression_paren MINUS NUMBER . $end
Action : Reduce rule [expression -> NUMBER] with ['1'] and goto state 64
Result : <function @ 0x104530af0> (<function FormulaParser.p_expression_num ...)

State  : 64
Stack  : expression_paren MINUS expression . $end
Action : Reduce rule [expression -> expression_paren MINUS expression] with [<function @ 0x104530940>,'-',<function @ 0x104530af0>] and goto state 2

```

Without

```
State  : 3
Stack  : expression_paren MINUS . LexToken(NUMBER,'1',1,15)
Action : Shift and goto state 5

State  : 5
Stack  : expression_paren MINUS NUMBER . $end
Action : Reduce rule [expression -> NUMBER] with ['1'] and goto state 31
Result : <function @ 0x13df02dc0> (<function FormulaParser.p_expression_num ...)

State  : 31
Stack  : expression_paren MINUS expression . $end
Action : Reduce rule [expression -> MINUS expression] with ['-',<function @ 0x13df02dc0>] and goto state 33
>>> UNARY
Result : <function @ 0x13df02ca0> (<function FormulaParser.p_expression_umi ...)

State  : 33
Stack  : expression_paren expression . $end
Action : Reduce rule [expression -> expression_paren expression] with [<function @ 0x13df02e50>,<function @ 0x13df02ca0>] and goto state 2
Result : <function @ 0x13df02c10> (<function FormulaParser.p_expression_imp ...)

```

Reply:
AFAIK this is specific enough to stop reducing to down to the `expression_paren expression` rule.

Reply:
This looks correct. I think this is a much better solution that what we had initially come up with, since I think messing with the precedence of the unary minus is not a good idea